feat: Added 'Allow Import (via Data Import Tool)' checkbox in Customize Form

marination opened the pull request:
- Checkbox will let users enable DocTypes in Data Import
- Document Type Link Field in Data Import will fetch Importable DocTypes from meta

![Screenshot 2019-09-27 at 1 14 54 PM](https://user-images.githubusercontent.com/25857446/65751766-04af0b00-e129-11e9-9a8c-7df8fb1de1f6.png)

marination committed:
feat: Added 'Allow Import (via Data Import Tool)' checkbox in Customize Form

Checkbox will let users enable DocTypes in Data Import
Document Type Link Field in Data Import will fetch Importable DocTypes from meta

surajshetty3416 commented on frappe/utils/user.py:
```diff
@@ -157,8 +157,11 @@ def build_permissions(self):
-			self.can_import = list(filter(lambda d: d in self.can_create,
-				frappe.db.sql_list("""select name from `tabDocType` where allow_import = 1""")))
+			docs = [x['name'] for x in frappe.get_all("DocType", "name")]
+			frappe.clear_cache()
```
Why? if you are getting cached meta then either use `frappe.get_meta(dt, cached=False)` or clear cache of just updated doctype eg. `frappe.cache().hdel('form_meta', 'User')` after updating from Customize Form.

**Note:** Clearing complete cache is not good for system performance

marination replied:
was unaware, alright

surajshetty3416 commented:
@marination any update?

marination committed:
fix: Importable docs via Customize Form set in redis cache

can_import list is set in redis cache
Data Import's set query will fetch from cache ad update list
